res/controllers/Traktor-Kontrol-S4-MK2-hid-scripts.js

```diff
@@ -1500,3 +1531,32 @@ TraktorS4MK2.onLoopEnabledChanged = function(value, group, key) {
   TraktorS4MK2.outputCallbackLoop(value, group, "loop_in");
   TraktorS4MK2.outputCallbackLoop(value, group, "loop_out");
 }
+
+TraktorS4MK2.onLoopSizeChanged = function(value, group, key) {
+  var deck = TraktorS4MK2.resolveDeckIfActive(group);
+  //deal with single digit values
+  if (value.toString().length === 1){
+    TraktorS4MK2.sendLoopSizeMessage(deck, '', value, false, false);
+  // values with two digits
+  } else if (value.toString().length === 2 ) {
+    TraktorS4MK2.sendLoopSizeMessage(deck, value.toString().split("")[0], value.toString().split("")[1], false, false);
+  // deal with fraction beats
+  } else if (1 > value > 0 ) {
+    if (value === 0.5){
+      TraktorS4MK2.sendLoopSizeMessage(deck, '-', 2, false, false);
+    } else if (value === 0.25){
+      TraktorS4MK2.sendLoopSizeMessage(deck, '-', 4, false, false);
+    } else if (value === 0.125){
+      TraktorS4MK2.sendLoopSizeMessage(deck, '-', 8, false, false);
+    } else {
+      TraktorS4MK2.sendLoopSizeMessage(deck, '-', 'n', false, false);
+    }
+  // deal with larger Loops
+  } else if (value.toString().length > 2){
+    TraktorS4MK2.sendLoopSizeMessage(deck, value.toString().split("")[0], value.toString().split("")[1], true, true);
```

Comment:
Maybe light up both dots in this case?

Reply:
That is what is happening here

Reply:
Oh, I see. Sorry, I didn't look at the code close enough before.